ebuild-writing/variables/text.xml

```diff
@@ -399,48 +399,66 @@ SRC_URI="https://example.com/files/${PV}.tar.gz -> ${P}.tar.gz"
-      <c>::gentoo</c> repository contains named groups of mirrors,
-      accessible through the <c>mirror://</c> pseudo-protocol.
-    </p>
-    <p>
-      As of EAPI 7, the format of <uri
-      link="https://projects.gentoo.org/pms/7/pms.html#x1-340004.4.2">the
-      thirdpartymirrors file</uri> is described in section 4.4.2 of
-      the Package Manager Specification (PMS). One might define a set
-      of &quot;example&quot; mirrors,
-    </p>
+<title>Third-party mirrors</title>
+<body>
+
+<p>
+If the items in <c>SRC_URI</c> are available on multiple third-party mirrors,
+and the same set of mirrors is shared across multiple ebuilds, then you
+don't have to repeatedly list each of them in every ebuild.
+The <c>profiles/thirdpartymirrors</c> file in the <c>::gentoo</c> repository
+contains named groups of mirrors, accessible through the <c>mirror://</c>
+pseudo-protocol.
+</p>
+
+<p>
+As of EAPI 7, the format of <uri
+link="https://projects.gentoo.org/pms/7/pms.html#x1-340004.4.2">the
+thirdpartymirrors file</uri> is described in section 4.4.2 of the Package
+Manager Specification (PMS). One might define a set of &quot;example&quot;
+mirrors,
+</p>
+
 <!--
   The following isn't ebuild code, but lang="ebuild" works, and no
   other language choice would be accurate either.
 -->
 <codesample lang="ebuild">
-example http://download.example.org/ ftp://ftp.example.org/
+example https://download.example.com https://mirror1.example.org/example
 </codesample>
-    <p>
-      that can afterwards be referenced through a <c>mirror://</c>
-      URL:
-    </p>
+
+<p>
+that can afterwards be referenced through a <c>mirror://</c> URL:
+</p>
+
 <codesample lang="ebuild">
 SRC_URI="mirror://example/${PN}/${P}.tar.gz"
 </codesample>
-    <warning>
-      If the sources for a package are neither mirror-restricted nor
-      fetch-restricted, then they will be <uri
-      link="::general-concepts/mirrors">mirrored onto Gentoo
-      infrastructure</uri> automatically; in that case, there is
-      little benefit to using this feature. There is a maintenance
-      burden associated with using a third-party mirror list: it must
-      be updated when new mirrors are born and when old mirrors
-      die. Please consider that when deciding whether or not to use
-      this feature.
-    </warning>
-  </body>
+
+<p>
+There are two valid cases for using <c>thirdpartymirrors</c>:
+</p>
+
+<ol>
+  <li>
+    providing multiple download locations for mirror- or fetch-restricted
+    packages,
+  </li>
+
+  <li>
+    dealing with upstreams that distribute their distfiles via a network
+    of mirrors without a primary download location or a bouncer service.
+  </li>
+</ol>
+
+<p>
+In any other case, the primary location must be used instead. The distfiles
+will be <uri link="::general-concepts/mirrors">mirrored onto Gentoo
+infrastructure</uri>; in that case, the benefit to using third-party mirror
+list does not outweigh the burden of maintaining it.
```

Comment:
I'm confused by what is meant in this part. If the purpose of this paragraph is to communicate the existence of the infra mirroring system, this is too quick and vague of a transition to talk about that. For example, the following makes for sense to me:

"""
If no mirroring opportunities are available, sources must be listed by their primary location in <c>SRC_URI</c> instead. These files will automatically be mirrored onto the Gentoo Infrastructure when applicable, which then gets replicated across many community source mirrors across the world. The Infra Project is in charge of maintaining the list of community mirrors. This process is transparent to the developers and no steps are needed to opt-in.
"""

Reply:
No, it's meant to supposed that otherwise you're not supposed to use `mirror://` even if there are 'opportunities' but just list the primary URI and let mirrors take care of mirroring.

Reply:
I've taken part of your sentence and integrated it into the basic idea. Not sure if it doesn't suggest that other files wouldn't be mirrored though.